airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceDebeziumOperations.kt

```diff
@@ -72,6 +75,22 @@ class MySqlSourceDebeziumOperations(
         val isDelete: Boolean = after.isNull
         // Use either `before` or `after` as the record data, depending on the nature of the change.
         val data: ObjectNode = (if (isDelete) before else after) as ObjectNode
+        // Turn string representations of numbers into BigDecimals.
+        for (field in stream.schema) {
+            when (field.type.airbyteSchemaType) {
+                LeafAirbyteSchemaType.INTEGER,
+                LeafAirbyteSchemaType.NUMBER -> {
+                    val textNode: TextNode = data[field.id] as? TextNode ?: continue
+                    val bigDecimal = BigDecimal(textNode.textValue()).stripTrailingZeros()
+                    data.put(field.id, bigDecimal)
+                }
+                LeafAirbyteSchemaType.JSONB -> {
+                    val textNode: TextNode = data[field.id] as? TextNode ?: continue
+                    data.set<JsonNode>(field.id, Jsons.readTree(textNode.textValue()))
+                }
+                else -> continue
+            }
```

Comment:
this feels like a bad abstraction here, the fact that we (the implementor of DebeziumOperations) have to iterate through the schema types. Just some food for thoughts

Reply:
Yeah, it's not great. Perhaps `Stream` should have pre-computed groupings of `Field`s per Airbyte type. Or per `FieldType`. Not sure.